src/toolkits/style_transfer/style_transfer.hpp

```diff
@@ -0,0 +1,77 @@
+/* Copyright © 2019 Apple Inc. All rights reserved.
+ *
+ * Use of this source code is governed by a BSD-3-clause license that can
+ * be found in the LICENSE.txt file or at
+ * https://opensource.org/licenses/BSD-3-Clause
+ */
+
+#ifndef __TOOLKITS_STYLE_TRANSFER_H_
+#define __TOOLKITS_STYLE_TRANSFER_H_
+
+#include <memory>
+
+#include <core/data/sframe/gl_sarray.hpp>
+#include <ml/neural_net/compute_context.hpp>
+#include <ml/neural_net/model_backend.hpp>
+#include <ml/neural_net/model_spec.hpp>
+#include <model_server/lib/extensions/ml_model.hpp>
+#include <toolkits/coreml_export/mlmodel_wrapper.hpp>
+#include <toolkits/style_transfer/style_transfer_data_iterator.hpp>
+
+namespace turi {
+namespace style_transfer {
+
+class EXPORT style_transfer : public ml_model_base {
+ public:
+  void init_options(const std::map<std::string, flexible_type>& opts) override;
+  size_t get_version() const override;
+  void save_impl(oarchive& oarc) const override;
+  void load_version(iarchive& iarc, size_t version) override;
+
+  void train(gl_sarray style, gl_sarray content,
+             std::map<std::string, flexible_type> opts);
+
+  BEGIN_CLASS_MEMBER_REGISTRATION("style_transfer")
+  IMPORT_BASE_CLASS_REGISTRATION(ml_model_base);
+  END_CLASS_MEMBER_REGISTRATION
+
+ protected:
+  virtual std::unique_ptr<data_iterator> create_iterator(
+      data_iterator::parameters iterator_params) const;
+
+  std::unique_ptr<data_iterator> create_iterator(gl_sarray style,
+                                                 gl_sarray content, bool repeat,
+                                                 int random_seed) const;
+
+  virtual std::unique_ptr<neural_net::compute_context> create_compute_context()
+      const;
+
+  virtual void init_train(gl_sarray style, gl_sarray content,
+                          std::map<std::string, flexible_type> opts);
+
+  virtual void perform_training_iteration();
```

Comment:
Actually, could we make these public and use the patterns I (just) established for OD? We want all the toolkits to eventually have a consistent API to support checkpointing, and we may as well bake it in from the beginning for new toolkits

https://github.com/apple/turicreate/blob/master/src/toolkits/object_detection/object_detector.hpp#L50